tests/unit/ui/components/TestOverlay.js

```diff
@@ -1,73 +1,141 @@
-    const root = render({ children: 'hello' });
-    expect(root.overlayContents).toBeTruthy();
-    expect(root.overlayContents.textContent).toContain('hello');
+    const html = '<div>overriding child div</div>';
+    const root = render({ children: html });
+    expect(
+      root
+        .children()
+        .at(1)
+        .props().children,
+    ).toEqual(html);
   });
 
   it('is hidden by default', () => {
     const root = render();
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+    expect(root).not.toHaveClassName('Overlay--visible');
   });
 
   it('is visible when the `visibleOnLoad` prop is passed', () => {
-    const root = render({ visibleOnLoad: true });
-    expect(root.overlayContainer.className).toContain('Overlay--visible');
+    const { store } = dispatchClientMetadata();
+
+    const root = render({
+      store,
+      visibleOnLoad: true,
+      uiState: { visible: false },
+    });
+
+    expect(root).not.toHaveClassName('Overlay--visible');
+
+    applyUIStateChanges({ root, store });
+
+    // Applying this 2 times here to mimic how uiState works in componentWillReceiveProps:
+    // The first time through, the visible state gets changed but these changes
+    // aren't shown in uiState till the 2 time through
+    applyUIStateChanges({ root, store });
+
+    expect(root).toHaveClassName('Overlay--visible');
   });
 
   it('hides when you click the background', () => {
-    const root = render({ visibleOnLoad: true });
-    Simulate.click(root.overlayBackground);
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+    const onClickBackgroundSpy = sinon.stub();
+    const root = render({
+      visibleOnLoad: true,
+      _onClickBackground: onClickBackgroundSpy,
+    });
+    const btn = root.find('.Overlay-background');
+    btn.simulate('click', createFakeEvent());
+    sinon.assert.called(onClickBackgroundSpy);
+    expect(root).not.toHaveClassName('Overlay--visible');
   });
 
   it('calls onEscapeOverlay when clicking the background', () => {
+    const clickEvent = createFakeEvent();
     const onEscapeOverlay = sinon.stub();
-    const root = render({ visibleOnLoad: true, onEscapeOverlay });
-    Simulate.click(root.overlayBackground);
+    const root = render({
+      visibleOnLoad: true,
+      onEscapeOverlay,
+    });
+    const btn = root.find('.Overlay-background');
+    btn.simulate('click', clickEvent);
+    const rootInstance = root.instance();
+    rootInstance.props._onClickBackground(clickEvent, rootInstance);
     sinon.assert.called(onEscapeOverlay);
   });
 
   it('is shown and hidden when `hide()` and `show()` are called', () => {
-    const root = render();
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
+
+    expect(root).not.toHaveClassName('Overlay--visible');
 
-    root.show();
-    expect(root.overlayContainer.className).toContain('Overlay--visible');
+    root.instance().show();
```

Comment:
I think you can get rid of the public `.show()`, `.hide()`, and `.toggle()` methods because, as far as I see, nothing is even using these methods. As I pointed out up above, this is a very un-React way to do things. It requires storing refs to components which we try not to do anymore, especially in production code. I think both `AddonReview` and `UserProfileEdit` are showing/hiding with the `visibleOnLoad` prop -- this should be enough.

Reply:
Right,  I wanted to do b/c I didn't even see it used. So cool, I'll remove this : ) thanks

Reply:
actually, we are calling the hide method in the clickbackground function so I'll leave that for now..

Reply:
Yes, it's OK for the component to call its own `.hide()` method internally.